numpy/core/src/multiarray/scalartypes.c.src

```diff
@@ -4111,37 +4099,32 @@ initialize_numeric_types(void)
     /**end repeat**/
 
 
-    /*
-     * These need to be coded specially because getitem does not
-     * return a normal Python type
-     */
-    PyLongDoubleArrType_Type.tp_as_number = &longdoubletype_as_number;
-    PyCLongDoubleArrType_Type.tp_as_number = &clongdoubletype_as_number;
 
     /**begin repeat
-     * #name = int, float, repr, str#
-     * #kind = tp_as_number->nb*2, tp*2#
+     * #char = ,c#
+     * #CHAR = ,C#
      */
 
-    PyLongDoubleArrType_Type.@kind@_@name@ = longdoubletype_@name@;
-    PyCLongDoubleArrType_Type.@kind@_@name@ = clongdoubletype_@name@;
-
-    /**end repeat**/
-
-
-#if !defined(NPY_PY3K)
-    /**begin repeat
-     * #name = long, hex, oct#
-     * #kind = tp_as_number->nb*3#
+    /*
+     * These need to be coded specially because longdouble/clongdouble getitem
+     * does not return a normal Python type
      */
+    @char@longdoubletype_as_number.nb_float = @char@longdoubletype_float;
+#if defined(NPY_PY3K)
+    @char@longdoubletype_as_number.nb_int  = @char@longdoubletype_long;
```

Comment:
This is deliberate, as nb_int should try to return `int` on py2, but `long` on py3. I'm not sure it makes a difference, but it's more consistent with how `longobject` is implemented.